build/docker.mk

```diff
@@ -97,50 +99,48 @@ configure-buildx:
 		docker buildx inspect --builder radius-builder --bootstrap; \
 	fi
 
-# defines a target for each image
-DOCKER_IMAGES := ucpd applications-rp controller
-
-$(foreach IMAGE,$(DOCKER_IMAGES),$(eval $(call generateDockerTargets,$(IMAGE),.,./deploy/images/$(IMAGE)/Dockerfile, go)))
-
-# multi arch container image targets for each binaries
-$(foreach IMAGE,$(DOCKER_IMAGES),$(eval $(call generateDockerMultiArches,$(IMAGE),.,./deploy/images/$(IMAGE)/Dockerfile)))
+# Define a target for each image with name and Dockerfile location
+APPS_MAP := ucpd:./deploy/images/ucpd \
+	applications-rp:./deploy/images/applications-rp \
+	controller:./deploy/images/controller \
+	testrp:./test/testrp \
+	magpiego:./test/magpiego
+
+# Function to extract name and Dockerfile from the app string
+define parseApp
+$(eval NAME := $(shell echo $(1) | cut -d: -f1))
+$(eval DOCKERFILE_PATH := $(shell echo $(1) | cut -d: -f2))
+endef
 
-# magpie comes from our test directory.
-$(eval $(call generateDockerTargets,magpiego,./test/magpiego/,./test/magpiego/Dockerfile))
+# This command will dynamically generate the targets for each image in the APPS_MAP list.
+$(foreach APP,$(APPS_MAP),$(eval $(call parseApp,$(APP)) $(call generateDockerTargets,$(NAME),.,$(DOCKERFILE_PATH)/Dockerfile,go)))
 
-# testrp comes from our test directory.
-$(eval $(call generateDockerTargets,testrp,./test/testrp/,./test/testrp/Dockerfile))
+# This command will dynamically generate the multi-arch targets for each image in the APPS_MAP list.
+$(foreach APP,$(APPS_MAP),$(eval $(call parseApp,$(APP)) $(call generateDockerMultiArches,$(NAME),.,$(DOCKERFILE_PATH)/Dockerfile)))
```

Comment:
Comments here are much much better compared to `build.mk`. Can you try to make that better too since you're changing it?